metadata-service/factories/src/main/java/com/linkedin/metadata/boot/UpgradeStep.java

```diff
@@ -0,0 +1,125 @@
+package com.linkedin.metadata.boot;
+
+import com.linkedin.common.AuditStamp;
+import com.linkedin.common.urn.Urn;
+import com.linkedin.data.DataMap;
+import com.linkedin.entity.EntityResponse;
+import com.linkedin.events.metadata.ChangeType;
+import com.linkedin.metadata.Constants;
+import com.linkedin.metadata.entity.EntityService;
+import com.linkedin.metadata.key.DataHubUpgradeKey;
+import com.linkedin.metadata.utils.EntityKeyUtils;
+import com.linkedin.metadata.utils.GenericRecordUtils;
+import com.linkedin.mxe.MetadataChangeProposal;
+import com.linkedin.upgrade.DataHubUpgradeRequest;
+import com.linkedin.upgrade.DataHubUpgradeResult;
+import java.net.URISyntaxException;
+import java.util.Collections;
+import lombok.extern.slf4j.Slf4j;
+
+
+@Slf4j
+public abstract class UpgradeStep implements BootstrapStep {
+  private static final Integer SLEEP_SECONDS = 120;
+
+  protected final EntityService _entityService;
+  private final String _version;
+  private final String _upgradeId;
+  private final Urn _upgradeUrn;
+
+  public UpgradeStep(EntityService entityService, String version, String upgradeId) {
+    this._entityService = entityService;
+    this._version = version;
+    this._upgradeId = upgradeId;
+    this._upgradeUrn = EntityKeyUtils.convertEntityKeyToUrn(new DataHubUpgradeKey().setId(upgradeId),
+        Constants.DATA_HUB_UPGRADE_ENTITY_NAME);
+  }
+
+  @Override
+  public void execute() throws Exception {
+    String upgradeStepName = name();
+
+    log.info(String.format("Attempting to run %s Upgrade Step..", upgradeStepName));
+    log.info(String.format("Waiting %s seconds..", SLEEP_SECONDS));
+
+    // Sleep to ensure deployment process finishes.
+    Thread.sleep(SLEEP_SECONDS * 1000);
+
+    if (hasUpgradeRan()) {
```

Comment:
how would you feel about doing the sleep after the check for whether or not the upgrade has run? There's no reason to wait around for 2 minutes on bootup of GMS if the step has already been run and we're going to early exit anyway. I've noticed a slowdown in GMS starting for myself locally and I'm wondering if this is the culprit (how it was before with the restore glossary indices step)

Reply:
Works for me!

Reply:
yeah agreed.

Reply:
Done